hikari/internal/ux.py

```diff
@@ -253,45 +254,62 @@ def supports_color(allow_color: bool, force_color: bool) -> bool:
     return color_support
 
 
-class HikariVersion(distutils.version.StrictVersion):
-    """Hikari-compatible strict version."""
+_VERSION_REGEX: typing.Final[re.Pattern[str]] = re.compile(r"^(\d+)\.(\d+)\.(\d+)(\.[a-z]+)?(\d+)?$", re.I)
```

Comment:
isn't re.Pattern not subscriptable on 3.8 (maybe typing.Pattern might be better for backwards compat here)?

Reply:
`typing.Pattern` has been deprecated as of Python 3.9 in favour of `re.Pattern`. I have tested it and 3.8 seems to support `re.Pattern`

Reply:
did you test that on 3.8

Reply:
Yep, my dev enviroment runs python 3.8.

Reply:
```py
Python 3.8.7 (tags/v3.8.7:6503f05, Dec 21 2020, 17:59:51) [MSC v.1928 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> import re
>>> re.Pattern[str]
Traceback (most recent call last):
  File "<pyshell#1>", line 1, in <module>
    re.Pattern[str]
TypeError: 'type' object is not subscriptable
```
also the docs state that typing.Pattern was deprecated in 3.9 because re.Pattern was made generic in 3.9, just like every other generic alias in typing like Mapping, MutableMapping, Sequence, MutableSequence, AbstractSet, Set, etc etc

Reply:
Oh, im guessing its `from __future__ import annotations` that is making it ignore it then. Because I have never had that error

Reply:
well we should be using typing.Pattern and if we were really using this approach consistently we'd be using collections.abc and the standard types (e.g. type, tuple) in place of the typing aliases.

Reply:
Thanks for spotting this issue :)